javascript/react/src/components/ArqueroDetailsList/__tests__/Columns/Columns.hooks.ts

```diff
@@ -0,0 +1,50 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type { ICheckboxProps, IColumn } from '@fluentui/react'
+import type ColumnTable from 'arquero/dist/types/table/column-table.js'
+import { useMemo, useState } from 'react'
+
+export function useCheckboxConfigs(table: ColumnTable | undefined): {
+	checkboxes: ICheckboxProps[]
+} {
+	const [checked, setChecked] = useState<string[]>(table?.columnNames() || [])
+	const checkboxes = useMemo(() => {
+		return (
+			table?.columnNames().map(column => ({
+				label: column,
+				checked: checked?.includes(column),
+				onChange: (
+					_ev?: React.FormEvent<HTMLElement | HTMLInputElement>,
+					chkd?: boolean,
```

Comment:
are we really saving that much by truncating this variable? Let's just call it `checked`

Reply:
Done